src/main/java/gov/nasa/pds/api/engineering/controllers/MyProductsApiBareController.java

```diff
@@ -93,6 +94,36 @@ public MyProductsApiBareController(ObjectMapper objectMapper, HttpServletRequest
 	        
     }
     
+    protected Products getLIDVIDs(List<String> lidvids, List<String> fields, HashSet<String> uniqueFields, boolean onlySummary) throws IOException
+    {
+    	String[] aFields = new String[fields == null ? 0 : fields.size()];
+    	if (fields != null)
+    	{
+    		for (int i = 0 ;  i < fields.size(); i++) aFields[i] = ElasticSearchUtil.jsonPropertyToElasticProperty(fields.get(i));
+    	}
+
+    	BoolQueryBuilder find_lidvids = QueryBuilders.boolQuery();
+    	Products results = new Products();
+    	SearchRequest request = new SearchRequest(this.esRegistryConnection.getRegistryIndex())
+    			.source(new SearchSourceBuilder().query(find_lidvids)
+    					.fetchSource(fields == null ? this.DEFAULT_ALL_FIELDS : aFields, this.DEFAULT_BLOB));
+
+    	for (String lidvid : lidvids) find_lidvids.should (QueryBuilders.termQuery ("lidvid", lidvid));
+    	for (SearchHit hit : this.esRegistryConnection.getRestHighLevelClient().search(request, RequestOptions.DEFAULT).getHits())
+    	{
+    		Map<String, Object> response = ElasticSearchUtil.elasticHashMapToJsonHashMap(hit.getSourceAsMap());
+    		uniqueFields.addAll(response.keySet());
+    		
+    		if (!onlySummary)
+    		{
+    			results.addDataItem (ElasticSearchUtil.ESentityProductToAPIProduct(
+    					this.objectMapper.convertValue(response, EntityProduct.class)));
+    			results.getData().get(results.getData().size()-1).setProperties(response);
+    		}
+    	}
+    	return results;
+    }
+
     protected Products getProducts(String q, int start, int limit, List<String> fields, List<String> sort, boolean onlySummary) throws IOException {
```

Comment:
You will also need to update this method, but that can be done in a later pull request.

Reply:
@tloubrieu-jpl 

Yes, most of the controllers need to be updated to become more efficient as #13 requires. Once we are concur on the function leveraging ES then the controllers will be updated.